js/modules/loki_message_api.js

```diff
@@ -2,9 +2,59 @@
 /* eslint-disable no-loop-func */
 /* global log, dcodeIO, window, callWorker, Whisper */
 
-const fetch = require('node-fetch');
+const nodeFetch = require('node-fetch');
 const _ = require('lodash');
 
+class HTTPError extends Error {
+  constructor(response) {
+    super(response.statusText);
+    this.name = 'HTTPError';
+    this.response = response;
+  }
+}
+
+class NotFoundError extends Error {
+  constructor() {
+    super('ENOTFOUND');
+    this.name = 'NotFoundError';
+  }
+}
+
+// A small wrapper around node-fetch which deserializes response
+const fetch = async (url, options = {}) => {
+  const timeout = 10000;
```

Comment:
Nit: for consistency with the following line? I can imagine a scenario where we would want different timeout for sending and retrieving.
```suggestion
  const timeout = options.timeout || 10000;
```
This is probably more explicit than just the `...options` at line 32.